### PR TITLE
feat(ingest/metabase): add ability to exclude other users collections

### PR DESCRIPTION
**Pull request by paguos**
Closes https://github.com/datahub-project/datahub/issues/10328

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by paguos** [comment]
Related to issue #10328

File: metadata-ingestion/src/datahub/ingestion/source/metabase.py [hunk 1]
```diff
@@ -79,6 +79,10 @@ class MetabaseConfig(DatasetLineageProviderConfigBase):
         default="public",
         description="Default schema name to use when schema is not provided in an SQL query",
     )
+    exclude_other_user_collections: str = Field(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/metabase.py:82]: this should be a bool type here

File: metadata-ingestion/src/datahub/ingestion/source/metabase.py [hunk 2]
```diff
@@ -209,6 +213,7 @@ def emit_dashboard_mces(self) -> Iterable[MetadataWorkUnit]:
         try:
             collections_response = self.session.get(
                 f"{self.config.connect_uri}/api/collection/"
+                f"?exclude-other-user-collections={self.config.exclude_other_user_collections}"
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/metabase.py:216]: and then we can json.dumps the bool value to convert it to `false` or `true`

**Review by hsheth2 — COMMENTED** [review]
One nit, otherwise looks good

File: metadata-ingestion/src/datahub/ingestion/source/metabase.py (hunk 1 above)
Comment (paguos) [reply metadata-ingestion/src/datahub/ingestion/source/metabase.py]: Makes sense

**Comment by paguos** [comment]
@hsheth2 thank you for the review! 

- I made the changes you requested
- Fixed the integration test

I amend my commit, so its difficult to see the changes (sorry).

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**